doc/services/llext/load.rst

```diff
@@ -0,0 +1,81 @@
+Loading extensions
+##################
+
+Once an extension is built and the ELF file is available, it can be loaded into
+the Zephyr application using the LLEXT API, which provides a way to load the
+extension into memory, access its symbols and call its functions.
+
+Loading an extension
+====================
+
+An extension may be loaded using any implementation of a :c:struct:`llext_loader`
+which has a set of function pointers that provide the necessary functionality
+to read the ELF data. A loader also provides some minimal context (memory)
+needed by the :c:func:`llext_load` function. An implementation over a buffer
+containing an ELF in addressable memory in memory is available as
+:c:struct:`llext_buf_loader`.
+
+The extensions are loaded with a call to the :c:func:`llext_load` function,
+passing in the extension name and the configured loader. Once that completes
+successfully, the extension is loaded into memory and is ready to be used.
+
+.. note::
+   When :ref:`User Mode <usermode_api>` is enabled, the extension will not be
+   included in any user memory domain. To allow access from user mode, the
+   :c:func:`llext_add_domain` function must be called.
+
+Accessing code and data
+=======================
+
+To interact with the newly loaded extension, the host application must use the
+:c:func:`llext_find_sym` function to get the address of the exported symbol.
+The returned ``void *`` can then be cast to the appropriate type and used.
+
+A wrapper for calling a function with no arguments is provided in
+:c:func:`llext_call_fn`.
+
+Cleaning up after use
+=====================
+
+The :c:func:`llext_unload` function must be called to free the memory used by
+the extension once it is no longer required. After this call completes, all
+pointers to symbols in the extension that were obtained will be invalid.
+
+Troubleshooting
+###############
```

Comment:
Minor nit: why is this section not inside the "Loading extensions" one? Or, if this is meant to be at the same level, why doesn't it live on a different file? I ask because if you look at the rendered result (https://builds.zephyrproject.io/zephyr/pr/75126/docs/services/llext/load.html), "Troubleshooting" is at the same as "API reference", "Configuration", but lives on the same file as "Loading extensions", while each other has its own file. 
![Screenshot_2024-07-01_11-23-11](https://github.com/zephyrproject-rtos/zephyr/assets/3177551/00f336d5-bf57-47d1-9892-4a06f7c58a77)

Reply:
That was initially a mistake, but I liked the result so I kept it on purpose. It makes that section visible from the top-level menu, for those looking for it, while still being part of the "Loading" section so that if you are reading the chapter you'll glance at it as well (doesn't hurt to be prepared! :slightly_smiling_face:). 

If that is not appropriate, I will move the content to a different file and perhaps leave just a reference here.